wordpress-mobile/WordPress-iOS
# Stats Revamp: Fix Insights cards not adapting to large device text size

Fixes #19017

## Description

Labels in Stats cards truncate or do not fit well when font size increases. 

## Solution

Go card by card and find the most appropriate solution: a) Go multiline if possible b) Limit the expansion of the label c) Allow the label to truncate

### Views & Visitors
- Allow cell title to go multiline
- Assign minimum amount of pace for title and details (weeks) button so they wouldn't shrink indefinitely using width constraints
- Allow date labels (this week / previous week) as well as views labels to shrink if they don't fit anymore
- Limit the size of labels in x and y axis

### Comments/Likes/Followers Total
- Allow count number to increase - support dynamic type
- Allow guide text to increase/decrease - support dynamic type for attributed string

### Most Popular Time
- Allow labels to go multiline 

### Latest Post Summary
- Allow column titles (View &  Likes & Comments) to shrink. They are usually one-word labels so going multiline is not a good option.

### All-Time / This Year / Today
- Allow labels to go multiline

### Posting Activity
- Allow month labels Aug/Sep/Oct increase to an extent
- Allow legend (fewer posts & more posts) to go multiline. Otherwise, they don't fit at all for smaller screen sizes, even without increased font size

## Testing instructions

1. Enable statsRevamp feature flag.
2. Open Stats.
3. Click the Settings icon and add all the cards.
4. Increase & Decrease dynamic font size.  
5. Notice the most important information - titles and numbers increasing and fitting the card when dynamic font size increases.

## Regression Notes

1. Potential unintended areas of impact

- Breaking UI for Stats with StatsRevampV2 disabled
- Breaking newly created UI

6. What I did to test those areas of impact (or what existing automated tests I relied on)

Manual testing on devices with different screen sizes and different dynamic type

7. What automated tests I added (or what prevented me from doing so)

None

PR submission checklist:


## Images & Videos

### Before

<img src="https://user-images.githubusercontent.com/4062343/193522213-a0ea47a5-3c09-4f72-b470-9bd1a45efe6b.png" alt="drawing" width="300"/>
<img src="https://user-images.githubusercontent.com/4062343/193522220-13bcc3c4-e4e4-42ec-80a0-628298df2f64.png" alt="drawing" width="300"/>
<img src="https://user-images.githubusercontent.com/4062343/193522223-996b9286-e862-498f-8512-85562dd1adad.png" alt="drawing" width="300"/>
<img src="https://user-images.githubusercontent.com/4062343/193522226-5efe49f8-8998-4882-97aa-48e669ee71e5.png" alt="drawing" width="300"/>
<img src="https://user-images.githubusercontent.com/4062343/193522228-94efe36a-4c6b-4198-815a-96dd0e3d2c0e.png" alt="drawing" width="300"/>

### After

<img src="https://user-images.githubusercontent.com/4062343/193522422-5cf234bf-951b-4758-80de-305ffe2e34d2.png" alt="drawing" width="300"/>
<img src="https://user-images.githubusercontent.com/4062343/193522424-455d3e53-aecd-419c-bee2-a6f85afe02f4.png" alt="drawing" width="300"/>
<img src="https://user-images.githubusercontent.com/4062343/193522430-e4ed84d8-960d-46cc-b695-2fdb43d7e0fc.png" alt="drawing" width="300"/>
<img src="https://user-images.githubusercontent.com/4062343/193522433-4bc4294d-3d21-428e-bc8e-2cf0020816b5.png" alt="drawing" width="300"/>
<img src="https://user-images.githubusercontent.com/4062343/193571016-772eeea8-58cb-46c5-9e72-87a532e89937.png" alt="drawing" width="300"/>
<img src="https://user-images.githubusercontent.com/4062343/193522428-6de19571-0f5d-4dec-bfd0-78840285cee7.png" alt="drawing" width="300"/>
